### ISSUE TITLE
Actionlint does not support secrets: inherit with workflow_call

### ISSUE THREAD

**Issue by FalconerTC** [rhysd/actionlint#138]
When using `workflow_call` (https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow), secrets can be specified individually or you can set `inherit` to send all secrets automatically. eg from Github docs. 

```
jobs:
  call-workflow-passing-data:
    uses: octo-org/example-repo/.github/workflows/reusable-workflow.yml@main
    with:
      username: mona
    secrets: inherit
```

This results in the following ActionLint errors in the calling workflow
```
 "secrets" section is scalar node but mapping node is expected [syntax-check]
  secrets: inherit
```

And the following in the receiving workflow for any undeclared secrets
```
property "vault_url" is not defined in object type {github_token: string; actions_step_debug: string; actions_runner_debug: string} [expression]
```

**Comment by prein** [comment]
I'm using 1.6.22 and getting this false positive as well.
It's not only `secrets: inherit` but also `permissions:`. I think it's only these two that differ from "regular" step (https://docs.github.com/en/actions/using-workflows/reusing-workflows#supported-keywords-for-jobs-that-call-a-reusable-workflow)
Example:
```
    - name: test
      permissions:
        checks: write
        contents: read
      secrets: inherit
      uses: ./.github/workflows/reusable-test.yaml
      with:
        foo: ${{ inputs.foo }}
```
throws:
```
[...] unexpected key "secrets" for "step" section. expected one of "continue-on-error", "env", "id", "if", "name", "run", "shell", "timeout-minutes", "uses", "with", "working-directory" [syntax-check]
   |
73 |         secrets: inherit
   |         ^~~~~~~~
[...] unexpected key "permissions" for "step" section. expected one of "continue-on-error", "env", "id", "if", "name", "run", "shell", "timeout-minutes", "uses", "with", "working-directory" [syntax-check]
   |
74 |         permissions:
   |         ^~~~~~~~~~~~
```
